src/widgets/chart/lv_chart.c

```diff
@@ -1244,6 +1288,115 @@ static void draw_series_bar(lv_obj_t * obj, lv_layer_t * layer)
+    int32_t pad_bottom = lv_obj_get_style_pad_bottom(obj, LV_PART_MAIN);
+    int32_t w = lv_obj_get_content_width(obj);
+    int32_t h  = lv_obj_get_content_height(obj);
+    lv_chart_series_t * ser;
+    uint32_t ser_cnt = lv_ll_get_len(&chart->series_ll);
+    if(ser_cnt == 0) {
+        return;
+    }
+
+    int32_t block_gap = lv_obj_get_style_pad_column(obj, LV_PART_MAIN);  /*Gap between the column on ~adjacent X*/
+    int32_t block_w = (w - ((chart->point_cnt - 1) * block_gap)) / chart->point_cnt;
+
+    int32_t border_w = lv_obj_get_style_border_width(obj, LV_PART_MAIN);
+    int32_t x_ofs = pad_left - lv_obj_get_scroll_left(obj) + border_w;
+    int32_t y_ofs = pad_bottom - lv_obj_get_scroll_top(obj) + border_w;
+
+    lv_draw_rect_dsc_t col_dsc;
+    lv_draw_rect_dsc_init(&col_dsc);
+    col_dsc.base.layer = layer;
+    lv_obj_init_draw_rect_dsc(obj, LV_PART_ITEMS, &col_dsc);
+    col_dsc.bg_grad.dir = LV_GRAD_DIR_NONE;
+    col_dsc.bg_opa = LV_OPA_COVER;
+
+    lv_area_t clip_area_ori = layer->_clip_area;
+
+    /*Go through all points*/
+    lv_area_t bar_full_area;
+    for(i = 0; i < chart->point_cnt; i++) {
+        col_dsc.base.id2 = i;
+        col_dsc.base.id1 = 0;
+
+        /*Get the total bar height. All segments (series) will be drawn in the height*/
+        int32_t v_sum_all = 0;
+        LV_LL_READ(&chart->series_ll, ser) {
+            if(ser->hidden) continue;
+            int32_t start_point = chart->update_mode == LV_CHART_UPDATE_MODE_SHIFT ? ser->start_point : 0;
+            int32_t p_act = (start_point + i) % chart->point_cnt;
+            int32_t v_act = ser->y_points[p_act];
+            if(ser->y_points[p_act] == LV_CHART_POINT_NONE) continue;
+
+            /* Skip negative values in stacked charts. Negative values are not supported
+             * in stacked charts as they cannot be visually represented in the stacking logic. */
+            if(v_act <= 0) {
+                LV_LOG_WARN("Stacked chart doesn't support negative values.");
+                continue;
+            }
+            v_sum_all += v_act;
+        }
+
+        int32_t total_bar_height = value_to_y(obj, lv_ll_get_head(&chart->series_ll), v_sum_all, h);
+        if(total_bar_height <= 0) continue;
+
+        if(chart->point_cnt <= 1) {
+            bar_full_area.x1 = obj->coords.x1 + x_ofs;
+        }
+        else {
+            bar_full_area.x1 = (int32_t)((int32_t)(w - block_w) * i) / (chart->point_cnt - 1) + obj->coords.x1 + x_ofs;
+        }
+        bar_full_area.x2 = bar_full_area.x1 + block_w - 1;
+        if(bar_full_area.x2 < clip_area_ori.x1) {
+            col_dsc.base.id1++;
+            continue;
+        }
+
+        if(bar_full_area.x1 > clip_area_ori.x2) break;
+
+        /*Draw the full_bar_area and set the clip area to clip the segments*/
+        bar_full_area.y2 = obj->coords.y2 + col_dsc.radius;
+        bar_full_area.y1 = bar_full_area.y2 - y_ofs - total_bar_height + 1;
```

Comment:
```suggestion
        bar_full_area.y1 = obj->coords.y2 - y_ofs - total_bar_height + 1;
```

I tested and found that the radius undesirably moves the bar's y1.